### PR TITLE
webhook scale host validation test

### PR DESCRIPTION
**Pull request by carolynhu**
https://github.com/rancher/rancher/issues/9024

[Webhook_ScaleHost_Test_plan.xlsx](https://github.com/rancher/validation-tests/files/1209534/Webhook_ScaleHost_Test_plan.xlsx)

### CODE REVIEW

File: tests/v3_validation/cattlevalidationtest/core/test_webhook_scale_host.py [hunk 1]
```diff
@@ -0,0 +1,919 @@
...
+    }
+
+    resp = create_webhook(env.id, data)
+    assert resp.status_code == 200
+    assert resp.url is not None
+    json_resp = json.loads(resp.content)
+    webhook_up_url = json_resp["url"]
+    webhook_up_id = json_resp["id"]
+
+    print "Webhook is " + repr(webhook_up_url)
```
Comment (cjellick) [inline tests/v3_validation/cattlevalidationtest/core/test_webhook_scale_host.py]: No print statements in tests

File: tests/v3_validation/cattlevalidationtest/core/test_webhook_scale_host.py [hunk 2]
```diff
@@ -0,0 +1,919 @@
...
+
+    # (1-2): Execute Webhook again and ensure the scale
+    # cannot be incremented beyond max
+    wh_resp = requests.post(webhook_up_url)
+    assert wh_resp.status_code == 400
+    json_resp = json.loads(wh_resp.content)
+    print "JSON response is:"
+    print json_resp
+
+    expected_response = "Error Cannot scale above provided max " \
```
Comment (cjellick) [inline tests/v3_validation/cattlevalidationtest/core/test_webhook_scale_host.py]: asserting on exact error messages is brittle and error prone. It is sufficient to assert on just the status_code

File: tests/v3_validation/cattlevalidationtest/core/test_webhook_scale_host.py [hunk 3]
```diff
@@ -0,0 +1,919 @@
...
+            "min": 1,
+            "max": 4
+        }
+    }
+
+    resp = create_webhook(env.id, data)
+    assert resp.status_code == 200
+    assert resp.url is not None
+    json_resp = json.loads(resp.content)
+    webhook_mostRecent_url = json_resp["url"]
```
Comment (cjellick) [inline tests/v3_validation/cattlevalidationtest/core/test_webhook_scale_host.py]: Use 'most_recent` instead. I know you based it on how the option is formatted, but consistency in the variable names is better

File: tests/v3_validation/cattlevalidationtest/core/test_webhook_scale_host.py [hunk 4]
```diff
@@ -0,0 +1,919 @@
...
+                                   )
+    assert last_host.state == 'removed'
+    assert base_host.state == 'active'
+    assert middle_host.state == 'active'
+
+    hosts = client.list_host()
+    host_scale_group = []
+    for host in hosts:
+        if host["hostTemplateId"] == base_host["hostTemplateId"]:
+            host = client.wait_success(host, timeout=DEFAULT_MACHINE_TIMEOUT)
```
Comment (cjellick) [inline tests/v3_validation/cattlevalidationtest/core/test_webhook_scale_host.py]: what is this wait_success line doing?

File: tests/v3_validation/cattlevalidationtest/core/test_webhook_scale_host.py [hunk 5]
```diff
@@ -0,0 +1,919 @@
...
+                                   last_host,
+                                   lambda x: x.state == 'removed',
+                                   lambda x: 'Host state is ' + x.state,
+                                   timeout=DEFAULT_MACHINE_TIMEOUT
+                                   )
+    assert last_host.state == 'removed'
+    assert base_host.state == 'active'
+    assert middle_host.state == 'active'
+
+    hosts = client.list_host()
```
Comment (cjellick) [inline tests/v3_validation/cattlevalidationtest/core/test_webhook_scale_host.py]: Can you add a filter to this like client.list_host(hostTemplateId=base_host["hostTemplateId"] and then skip the for loop below?

Comment (carolynhu) [inline tests/v3_validation/cattlevalidationtest/core/test_webhook_scale_host.py]: I used it before, but the test will error prone, I think it's better to check the state of the host

**Comment by carolynhu** [comment]
Good to test when rancher 2.0 is stable

**Comment by cjellick** [comment]
Very stale. Closing. Thanks! Please open a new PR if there is still a relevant change.

**Pull request closed without merging**